box.sequence: sequence_data recovery after reset (#10940)
tarantool/tarantool: changelogs/unreleased/gh_9871_sequence_data_recovery_after_reset.md

## bugfix/core

* Fixed a bug when the `sequence:reset()` call result was not saved

>   in WAL and snapshot and thus was not recovered after the server restart

Author: Done. Now the commit message and changelog don't appeal to the WAL entries directly.

> * Fixed a bug that resulted in `sequence:reset()` call result not saved
>   in WAL and snapshot and thus not recovered after the server restart
>   (gh-9871).

Reviewer: ```suggestion
* Fixed a bug when the `sequence:reset()` call result was not saved
  in WAL and snapshot and thus was not recovered after the server restart
  (gh-9871).
```

Author: Thanks! Fixed.
